### PR TITLE
25-server-database

### PR DESCRIPTION
**Pull request by rikyiso01**

### CODE REVIEW

File: backend/server_db/sql_scripts/create_db.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+CREATE TABLE StudentsGroupMember (
+    student_id smallint PRIMARY KEY REFERENCES Student(student_id) ON UPDATE CASCADE,
+    students_group_id serial REFERENCES StudentsGroup(id) ON UPDATE CASCADE
+);
+
+CREATE TABLE JoinRequest (
+    join_request_id serial PRIMARY KEY,
+    group_id serial REFERENCES StudentsGroup(id) ON UPDATE CASCADE,
+    student_id integer REFERENCES Student(student_id) ON UPDATE CASCADE,
+    join_request_status varchar(20)
```
Comment (nomun19) [inline backend/server_db/sql_scripts/create_db.sql:34]: - add NOT NULL constraint for group_id, student_id

File: backend/server_db/sql_scripts/create_db.sql [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+);
+
+CREATE TABLE FbToken (
+    fb_token_id serial PRIMARY KEY,
+    token varchar(200),
+    student_id int REFERENCES Student(student_id) ON UPDATE CASCADE
```
Comment (nomun19) [inline backend/server_db/sql_scripts/create_db.sql:40]: - add NOT NULL constraint for student_Id and token

File: backend/server_db/sql_scripts/create_db.sql [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    fb_token_id serial PRIMARY KEY,
+    token varchar(200),
+    student_id int REFERENCES Student(student_id) ON UPDATE CASCADE 
+);
+
+CREATE TABLE Notification (
+    notification_id serial PRIMARY KEY,
+    student_id integer REFERENCES Student(student_id) ON UPDATE CASCADE,
+    join_request_id bigint REFERENCES JoinRequest(join_request_id) ON UPDATE CASCADE,
+    notification_type varchar(20)
```
Comment (nomun19) [inline backend/server_db/sql_scripts/create_db.sql:47]: - add not null constraint for student_id, join_request_id, notification type

**Review by nomun19 — COMMENTED** [review]
Generally, constraints need to be added for each table, and the StudentsGroupMember table needs to be fixed

**Review by AlexandruVasile — COMMENTED** [review]
We fixed the problems with db queries and we added a Makefile to manage the migration and the starting/stopping of postgres. The dockerfile is related to an image about migration.

**Pull request merged**